backend/tests/v5/unit/models/tickets.test.js

```diff
@@ -163,99 +168,113 @@ const testGetAllTickets = () => {
-			const teamspace = generateRandomString();
-			const ticketId = generateRandomString();
-			const propToUpdate = generateRandomString();
-			const data = { propToUpdate, properties: { propToUpdate }, modules: { module: { propToUpdate } } };
+			const oldPropvalue = generateRandomString();
+			const newPropValue = generateRandomString();
+			const oldTicket = { _id: generateRandomString(), properties: { propToUpdate: oldPropvalue } };
+			const data = { properties: { propToUpdate: newPropValue, 'Updated at': date } };
 			const fn = jest.spyOn(db, 'updateOne').mockResolvedValueOnce(undefined);
 
-			await Ticket.updateTicket(teamspace, ticketId, data);
+			await Ticket.updateTicket(teamspace, project, model, oldTicket, data, author);
 
 			expect(fn).toHaveBeenCalledTimes(1);
-			expect(fn).toHaveBeenCalledWith(teamspace, ticketCol, { _id: ticketId },
+			expect(fn).toHaveBeenCalledWith(teamspace, ticketCol, { _id: oldTicket._id },
 				{
 					$set: {
-						'modules.module.propToUpdate': propToUpdate,
-						propToUpdate,
-						'properties.propToUpdate': propToUpdate,
+						'properties.propToUpdate': newPropValue,
+						'properties.Updated at': date,
 					},
 				});
+			expect(EventsManager.publish).toHaveBeenCalledTimes(1);
+			expect(EventsManager.publish).toHaveBeenCalledWith(events.MODEL_TICKET_UPDATE, {
+				teamspace,
+				project,
+				model,
+				ticket: oldTicket._id,
+				author,
+				date,
+				from: { properties: { propToUpdate: oldPropvalue } },
+				to: { properties: { propToUpdate: newPropValue } },
+			});
 		});
 
-		test('should update the ticket to unset properties', async () => {
-			const teamspace = generateRandomString();
-			const ticketId = generateRandomString();
+		test('should update the ticket to set modules', async () => {
+			const oldPropvalue = generateRandomString();
+			const newPropValue = generateRandomString();
 			const data = {
-				propToUnset: null,
-				properties: { propToUnset: null },
-				modules: { module: { propToUnset: null } },
+				properties: { 'Updated at': date },
+				modules: { module: { propToUpdate: newPropValue } },
 			};
+			const oldTicket = { _id: generateRandomString(), modules: { module: { propToUpdate: oldPropvalue } } };
 			const fn = jest.spyOn(db, 'updateOne').mockResolvedValueOnce(undefined);
 
-			await Ticket.updateTicket(teamspace, ticketId, data);
+			await Ticket.updateTicket(teamspace, project, model, oldTicket, data, author);
 
 			expect(fn).toHaveBeenCalledTimes(1);
-
-			expect(fn).toHaveBeenCalledWith(teamspace, ticketCol, { _id: ticketId },
+			expect(fn).toHaveBeenCalledWith(teamspace, ticketCol, { _id: oldTicket._id },
 				{
-					$unset: {
-						'modules.module.propToUnset': 1,
-						propToUnset: 1,
-						'properties.propToUnset': 1,
+					$set: {
+						'modules.module.propToUpdate': newPropValue,
+						'properties.Updated at': date,
 					},
 				});
+			expect(EventsManager.publish).toHaveBeenCalledTimes(1);
```

Comment:
can we add a test to check the publish isn't called if something errored?

just mock the `updateOne` function to throw